### PR TITLE
Qdrant

### PR DESCRIPTION
**Pull request by CaptainLevi0007**
### Description

Implemented integration with Qdrant DB: This commit introduces the integration of the Qdrant database into our open-source project.

### Test Plan

The unit tests for all the functions I've written are structured as functions that begin with `test_`. These tests can be found in the `tests` directory.

### Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs update

### Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have read the contributing guide and my code conforms to the guidelines.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have added the required tests.

### CODE REVIEW

File: superagi/vector_store/qdrant.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+from superagi.vector_store.base import VectorStore
+from superagi.vector_store.document import Document
+
+DictFilter = Dict[str, Union[str, int, bool, dict, list]]
+MetadataFilter = Union[DictFilter, common_types.Filter]
+
+def create_qdrant_client(
+) -> QdrantClient:
+    client = QdrantClient(host="localhost", port=6333)
```
Comment (TransformerOptimus) [inline superagi/vector_store/qdrant.py:19]: this cant be hardcoded. You need to get this from config. Should have new config QDRANT_HOST and QDRANT_PORT

File: superagi/vector_store/qdrant.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+            )
+
+            collected_ids.extend(id_batch)
+
+        return collected_ids
+
+    def get_matching_text(
+            self,
+            embedding: List[float] = None,
+            k: int = 4,
```
Comment (TransformerOptimus) [inline superagi/vector_store/qdrant.py:79]: what is k? is it quadrant concept?

Comment (CaptainLevi0007) [reply superagi/vector_store/qdrant.py:79]: top k results to search

File: superagi/vector_store/vector_factory.py [hunk 3]
```diff
@@ -61,4 +64,18 @@ def get_vector_storage(cls, vector_store: VectorStoreType, index_name, embedding
             )
             return weaviate.Weaviate(client, embedding_model, index_name, 'text')
 
+        if vector_store == VectorStoreType.QDRANT:
+
+            client = qdrant.create_qdrant_client()
+            try:
+                client.get_collection(collection_name=index_name)
+            except UnexpectedResponse as e:
```
Comment (TransformerOptimus) [inline superagi/vector_store/vector_factory.py:72]: you have to check and create. This is exception driven logic not right.

File: superagi/vector_store/vector_factory.py [hunk 4]
```diff
@@ -61,4 +64,18 @@ def get_vector_storage(cls, vector_store: VectorStoreType, index_name, embedding
             )
             return weaviate.Weaviate(client, embedding_model, index_name, 'text')
 
+        if vector_store == VectorStoreType.QDRANT:
```
Comment (TransformerOptimus) [inline superagi/vector_store/vector_factory.py:67]: add unit test for vector factory

File: superagi/vector_store/vector_factory.py (hunk 3 above)
Comment (TransformerOptimus) [reply superagi/vector_store/vector_factory.py:72]: Moe this entire creation logic inside Qdrant class

**Review by TransformerOptimus — APPROVED** [review]

File: superagi/vector_store/vector_factory.py (hunk 4 above)
Comment (CaptainLevi0007) [reply superagi/vector_store/vector_factory.py:66]: https://github.com/TransformerOptimus/SuperAGI/pull/742

**Pull request merged**